tests/storage/zfsvol/test_zfsvol_sr.py

```diff
@@ -19,7 +19,8 @@ class TestZfsvolSRCreateDestroy:
     and VM import.
     """
 
-    def test_create_and_destroy_sr(self, host, sr_disk_wiped):
+    def test_create_and_destroy_sr(self, host_with_zfsvol, sr_disk_wiped):
```

Comment:
AFAIK the order of fixtures determines which one is executed first. Here, we can ask ourselves: what do we want to fail first? The answer is: `sr_disk_wiped`. If the SR is missing, there's no point in starting to install RPM packages through the `host_with_zfsvol` fixture.

Thus I'd swap both parameters.

Reply:
yes but where is this function called? Because all others tests are doing the same so I guess that there is some magic somewhere that calls it and we cannot do what we want?

Reply:
if other tests do the same, let's keep as is. We can improve them all together later if needed.

Reply:
> yes but where is this function called? 

pytest collects the fixtures, orders them, and then initializes them depending on their scope and what tests require them.

Reply:
What I observed additionnally is that when all other conditions have been treated by pytest and fixtures are at the same order and remain to be sorted, it looks like it then it oses the order in which it saw them. Could be an internal implementation detail and subject to change.